utils/regions_countries.py

```diff
@@ -0,0 +1,41 @@
+# /utils/country_feature_collection.py
+from django.contrib.gis.geos import GEOSGeometry
+from django.db.models import Q
+from areas.models import Area
+import json
+
+def get_regions_countries():
+    
+    # Uncomment the following code to regenerate regions_countries.json based on any boundary modifications
+    
+    # regions = Area.objects.all().filter((Q(type='predefined'))).values('id','type','title','geojson')
+    # countries = Area.objects.all().filter((Q(type='country'))).values('id','type','title','ccodes','geojson')
+    # country_codes = {country['id']: country['ccodes'] for country in countries}
+    # country_geometries = {country['id']: GEOSGeometry(json.dumps(country['geojson'])) for country in countries}
+    #
+    # # TODO: This code block would be unnecessary if all predefined areas had populated ccodes in database
+    # for region in regions:
+    #     region_geojson = GEOSGeometry(json.dumps(region['geojson']))
+    #     intersecting_ccodes = []
+    #
+    #     # Check intersection with each country
+    #     for country_id, country_geometry in country_geometries.items():
+    #         if region_geojson.intersects(country_geometry):
+    #             intersecting_ccodes.extend(country_codes[country_id])
+    #
+    #     region['ccodes'] = intersecting_ccodes
+    #
+    # # Transform lists for grouping in dropdown
+    # regions = [{'id': region['id'], 'text': region['title'], 'ccodes': region['ccodes']} for region in sorted(regions, key=lambda x: x['title'])]
+    # countries = [{'id': country['ccodes'][0], 'text': country['title']} for country in sorted(countries, key=lambda x: x['title'])]
+    #
+    # dropdown_data = [{'text':'Regions', 'children': list(regions)}, {'text': 'Countries', 'children': list(countries)}]
+    
+    # # Save the regenerated data to the JSON file
+    # with open('static/js/regions_countries.json', 'w') as json_file:
+    #     json.dump(dropdown_data, json_file, default=str)
+    
+    with open('static/js/regions_countries.json', 'r') as file:
```

Comment:
this file in static is not tracked - media/data is where several static files have lived typically

Reply:
Ah, OK - will you move it and update the code or shall I?

Reply:
I don't have it, as static is not tracked - please send it and I'll include it

Reply:
[regions_countries.json](https://github.com/WorldHistoricalGazetteer/whg3/files/14075511/regions_countries.json)